lib/codegen/src/write.rs

```diff
@@ -251,27 +270,27 @@ fn type_suffix(func: &Function, inst: Inst) -> Option<Type> {
     Some(rtype)
 }
 
-// Write out any value aliases appearing in `inst`.
-fn write_value_aliases(w: &mut Write, func: &Function, inst: Inst, indent: usize) -> fmt::Result {
-    for &arg in func.dfg.inst_args(inst) {
-        let resolved = func.dfg.resolve_aliases(arg);
-        if resolved != arg {
-            writeln!(w, "{1:0$}{2} -> {3}", indent, "", arg, resolved)?;
-        }
+// Write out any aliases to the given target
+fn write_value_aliases(
+    w: &mut Write,
+    aliases: &EntityMap<Value, Vec<Value>>,
+    target: Value,
+    indent: usize,
+) -> fmt::Result {
+    for a in &aliases[target] {
+        writeln!(w, "{1:0$}{2} -> {3}", indent, "", a, target)?;
```

Comment:
Could you make this look up the immediate target of the alias, rather than the fully resolved one? We want to print the IR as it is, without any simplifications.

Aliases to aliases should be rare, and in some cases could indicate a problem, so when they do occur, it's useful to see them. And, it's useful to be able to print the IR and read it back in without anything getting canonicalized.

Reply:
Pushed a squashme. Single-level alias following was defined in a `_for_parser` method, so I moved it to a normal method now that write.rs needs it too. I also had to use `make_value_alias_for_parser` in tests because it looks like only the parser can create an alias to an alias.

Reply:
We almost always do want to `resolve_aliases` though :-). I think the only cases where we don't are when we're serializing and deserializing the IR in some way. So, what if we left these routines as-is and renamed them from "for_parser" to "for_serialization"? Then we can call them from write.rs or other potential serialize/deserialize routines.

Reply:
OK, I did it that way